fl4health/utils/dataset.py

```diff
@@ -103,3 +102,55 @@ def __getitem__(self, index: int) -> Tuple[Dict[str, torch.Tensor], torch.Tensor
     def __len__(self) -> int:
         first_key = list(self.data.keys())[0]
         return len(self.data[first_key])
+
+
+class SyntheticDataset(TensorDataset):
+    def __init__(
+        self,
+        data: torch.Tensor,
+        targets: torch.Tensor,
+    ):
+        """
+        A dataset for synthetically created data strictly in the form of pytorch tensors.
+        Args:
+            data (torch.Tensor): Data tensor with first dimension corresponding to the number of datapoints
+            targets (torch.Tensor): Target tensor with first dimension corresponding to the number of datapoints
+        """
+        assert data.shape[0] == targets.shape[0]
+        self.data = data
+        self.targets = targets
+
+    def __getitem__(self, index: int) -> Tuple[torch.Tensor, torch.Tensor]:
+        assert self.targets is not None
+
+        data, target = self.data[index], self.targets[index]
+        return data, target
+
+    def __len__(self) -> int:
+        return len(self.data)
+
+
+D = TypeVar("D", bound=Union[TensorDataset, DictionaryDataset])
+
+
+def select_by_indices(dataset: D, selected_indices: torch.Tensor) -> D:
```

Comment:
Is there a specific reason we are moving select_by_indices out of LabelBasedSampler?

Reply:
Oh I see you want to also use it in partitioner. 👍

Reply:
Good question. I extracted the function because it doesn't use any class specific variables and it's actually agnostic of the fact that you're using any kind of label based sampler. That is, this can be used by any type of sampler (say random sampler for a trivial example) as long as you generate indicates to select in the dataset.